src/components/tooltips-hoc/TooltipHOC.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import {DropPod} from './DropPod';
+
+export interface ITooltipHOCProps {
+    buttonContainer?: React.HTMLAttributes<HTMLDivElement>;
+    listContainer?: React.HTMLAttributes<HTMLDivElement>;
+    renderOpenButton: (onClick: () => void) => React.ReactNode;
+    closeOnClick?: boolean;
+    positions?: string[];
+    selector?: string;
+}
+
+export interface ITooltipHOCState {
+    isOpen: boolean;
+}
+
+export class TooltipHOC extends React.PureComponent<ITooltipHOCProps, ITooltipHOCState> {
+    readonly button: React.RefObject<HTMLDivElement>;
+
+    static defaultProps: Partial<ITooltipHOCProps> = {
+        closeOnClick: true,
+    };
+
+    constructor(props: ITooltipHOCProps, state: ITooltipHOCState) {
+        super(props, state);
+
+        this.button = React.createRef();
+        this.state = {
+            isOpen: false,
+        };
+    }
+
+    componentWillMount() {
+        document.addEventListener('mousedown', this.handleDocumentClick);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleDocumentClick);
+    }
+
+    render() {
+        return (
+            <>
+                <div
+                    ref={this.button}
+                    {...this.props.buttonContainer}
+                >
+                    {this.props.renderOpenButton(() => this.onClick())}

```

Comment:
By using `() => this.onClick()`, you will create a new handler on each render.
To avoid that, you can simply use `this.props.renderOpenButton(this.onClick)` here and add `this.onClick.bind(this);` in the constructor.